pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -367,6 +367,13 @@ func (asw *actualStateOfWorld) addVolume(
 			devicePath:         devicePath,
 		}
 		asw.attachedVolumes[volumeName] = volumeObj
+	} else {
+		// If volume object already exists, update the fields such as device path
+		volumeObj.devicePath = devicePath
+		volumeObj.spec = volumeSpec
```

Comment:
This will not update the `devicePath`/`volumeSpec` for `asw.attachedVolumes[volumeName]`. It only updates the temporary `volumeObj` object. To update the value in the map, you have to write the temp object back:

```GO
		asw.attachedVolumes[volumeName] = volumeObj
```